### PR TITLE
Base64 Analyser

### PR DESCRIPTION
**Pull request by martin-bucinskas**
# Description
Implementation of a base64 analyser.

 - Created a simple class which extends RegexAnalyzer class.
 - Added unit tests with some edge cases.
 - Fixed regex to only match base64 encoded strings and not empty strings.

## Issue
[https://github.com/d-Rickyy-b/pastepwn/issues/120](https://github.com/d-Rickyy-b/pastepwn/issues/120)

### CODE REVIEW

**Comment by DaRuudii** [comment]
> 
> 
> @d-Rickyy-b I'm not quite sure what the codacy-bot means by the complexity increasing per file:
> 
> `Complexity increasing per file pastepwn/analyzers/base64analyzer.py 1`
> 
> Is this due to me breaking the strings up so it doesn't cap the 80 char limit?

The complexity increase just indicates that this file has the complexity of 1 which comes from the file being newly addded. (my guess, don't know if i am correct)

**Comment by d-Rickyy-b** [comment]
> 
> 
> @d-Rickyy-b I'm not quite sure what the codacy-bot means by the complexity increasing per file:
> 
> `Complexity increasing per file pastepwn/analyzers/base64analyzer.py 1`
> 
> Is this due to me breaking the strings up so it doesn't cap the 80 char limit?

No worries :) 

> Codacy uses cyclomatic complexity to identify complex files in your project. 
> 
> Cyclomatic Complexity is a software metric used to indicate the complexity of a program. It is a quantitative measure of the number of linearly independent paths through a program's source code. 

That means that a complexity of 1 is actually very good.

File: pastepwn/analyzers/base64analyzer.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# -*- coding: utf-8 -*-
+from .regexanalyzer import RegexAnalyzer
+
+
+class Base64Analyzer(RegexAnalyzer):
+    """Analyzer to match base64 encoding via regex"""
+    name = "Base64Analyzer"
+
+    def __init__(self, actions):
+        regex = r"^(?:[A-Za-z0-9+\/]{4})+(?:[A-Za-z0-9+\/]{2}==|[A-Za-z0-9+\/]{3}=)?$"
```
Comment (d-Rickyy-b) [inline pastepwn/analyzers/base64analyzer.py]: I am not sure if matching the beginning and the end of the string is good. I know that it is the same regex as in my description, I just want to leave this here to leave a reminder.

Comment (martin-bucinskas) [reply pastepwn/analyzers/base64analyzer.py]: I've changed it to what you have provided and changed the first instance of `*` to `+` to match one or more to prevent regex matching empty strings.

The end of the string needs to be matched too as the end will change depending upon the padding. I will give it another look later tonight and maybe come up with a more robust solution, but this regex hasn't failed yet during testing and on quite large data too.
